File: app/src/main/java/pl/kwako/metroid_map/DefaultSettings.java
package pl.kwako.metroid_map;

class DefaultSettings implements Settings {
    private static final int ROOMS_SIZE_X = 30;
    private static final int ROOMS_SIZE_Y = 30;
    private static final int ROOM_IMG_WIDTH = 256;
    private static final int ROOM_IMG_HEIGHT = 240;
    private static final int DEFAULT_WINDOW_WIDTH = 960;
    private static final int DEFAULT_WINDOW_HEIGHT = 960;

    @Override
    public int roomsSizeX() {
        return ROOMS_SIZE_X;
    }

    @Override
    public int roomsSizeY() {
        return ROOMS_SIZE_Y;
    }

    @Override
    public int roomImgWidth() {
        return ROOM_IMG_WIDTH;
    }

    @Override
    public int roomImgHeight() {
        return ROOM_IMG_HEIGHT;
    }

    @Override
    public int defaultWindowWidth() {
        return DEFAULT_WINDOW_WIDTH;
    }

    @Override
    public int defaultWindowHeight() {
        return DEFAULT_WINDOW_HEIGHT;
    }
}

File: app/src/main/kotlin/pl/kwako/metroid_map/DefaultSettings.kt
package pl.kwako.metroid_map

data class DefaultSettings(
        override val roomsSizeX: Int = 30,
        override val roomsSizeY: Int = 30,
        override val roomImgWidth: Int = 256,
        override val roomImgHeight: Int = 240,
        override val defaultWindowWidth: Int = 960,
        override val defaultWindowHeight: Int = 960
) : Settings {
    override fun roomRangeX(): IntRange = 0 until roomsSizeX
    override fun roomRangeY(): IntRange = 0 until roomsSizeY
}